test/hotspot/jtreg/gc/shenandoah/TestPeriodicGC.java

```diff
@@ -45,14 +46,39 @@ public static void testWith(String msg, boolean periodic, String... args) throws
         OutputAnalyzer output = ProcessTools.executeLimitedTestJava(cmds);
 
         output.shouldHaveExitValue(0);
-        if (periodic && !output.getOutput().contains("Trigger: Time since last GC")) {
+        if (periodic && !output.getOutput().contains("Trigger (GLOBAL): Time since last GC")) {
             throw new AssertionError(msg + ": Should have periodic GC in logs");
         }
-        if (!periodic && output.getOutput().contains("Trigger: Time since last GC")) {
+        if (!periodic && output.getOutput().contains("Trigger (GLOBAL): Time since last GC")) {
             throw new AssertionError(msg + ": Should not have periodic GC in logs");
         }
     }
 
+    public static void testGenerational(boolean periodic, String... args) throws Exception {
+        String[] cmds = Arrays.copyOf(args, args.length + 2);
+        cmds[args.length] = TestPeriodicGC.class.getName();
+        cmds[args.length + 1] = "test";
+        ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(cmds);
+
+        OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.shouldHaveExitValue(0);
+        if (periodic) {
+            if (!output.getOutput().contains("Trigger (YOUNG): Time since last GC")) {
+                throw new AssertionError("Generational mode: Should have periodic young GC in logs");
+            }
```

Comment:
Here, and in other tests, use `output.shouldContain(...)`

Reply:
https://bugs.openjdk.org/browse/JDK-8338695